src/services/structures/controllers/updateDemandeConseillerRefusAvisAdmin.ts

```diff
@@ -0,0 +1,129 @@
+import { Application } from '@feathersjs/express';
+import { Response } from 'express';
+import { ObjectId } from 'mongodb';
+import { IRequest } from '../../../ts/interfaces/global.interfaces';
+import { action } from '../../../helpers/accessControl/accessList';
+import service from '../../../helpers/services';
+import mailer from '../../../mailer';
+import { IStructures, IUser } from '../../../ts/interfaces/db.interfaces';
+import { PhaseConventionnement } from '../../../ts/enum';
+import {
+  refusCandidaturePosteConseiller,
+  refusCandidaturePosteConseillerPrefet,
+} from '../../../emails';
+
+const updateDemandeConseillerRefusAvisAdmin =
+  (app: Application) => async (req: IRequest, res: Response) => {
+    const idStructure = req.params.id;
+    if (!ObjectId.isValid(idStructure)) {
+      res.status(400).json({ message: 'Id incorrect' });
+      return;
+    }
+
+    try {
+      const structure: IStructures = await app
+        .service(service.structures)
+        .Model.accessibleBy(req.ability, action.read)
+        .findOne({
+          _id: new ObjectId(idStructure),
+          coordinateurCandidature: false,
+          statut: 'CREEE',
+        });
+      if (!structure) {
+        res.status(404).json({ message: "La structure n'existe pas" });
+        return;
+      }
+      const structureUpdated = await app
+        .service(service.structures)
+        .Model.accessibleBy(req.ability, action.update)
+        .findOneAndUpdate(
+          {
+            _id: structure._id,
+          },
+          {
+            $set: {
+              statut: 'REFUS_COSELEC',
+              coselecAt: new Date(),
+            },
+            $push: {
+              coselec: {
+                nombreConseillersCoselec: 0,
+                avisCoselec: 'NÉGATIF',
+                insertedAt: new Date(),
+                phaseConventionnement: PhaseConventionnement.PHASE_2,
+              },
+            },
+          },
+          {
+            new: true,
+          },
+        );
+
+      if (!structureUpdated) {
+        res
+          .status(404)
+          .json({ message: "La structure n'a pas été mise à jour" });
+        return;
+      }
+
+      const prefets: IUser[] = await app
+        .service(service.users)
+        .Model.accessibleBy(req.ability, action.read)
+        .find({
+          roles: { $in: ['prefet'] },
+          departement: structure.codeDepartement,
+        })
+        .select({ _id: 0, name: 1 });
+
+      const mailerInstance = mailer(app);
+      if (prefets.length > 0) {
```

Comment:
for of bounce ?

Reply:
on n'a pas eu besoin pour les coordinateurs, il y a pas beaucoup préfet par département